src/core/localization/extractLinkAndLabelFromMarkdown.ts

```diff
@@ -0,0 +1,34 @@
+import { fromMarkdown } from 'mdast-util-from-markdown';
+import type { Link } from 'mdast-util-from-markdown/lib';
+
+/**
+ * Function for extracting an array of links from a markdown string
+ * @param markdownString input markdown string
+ * @returns array of { link, label } objects
+ */
+
+export type LinkAndLabel = {
+  link?: string;
+  label?: string;
+};
+
+export const extractLinkAndLabels = (markdownString: string): LinkAndLabel[] => {
+  const result: LinkAndLabel[] = [];
+  const paragraph = fromMarkdown(markdownString).children[0];
+  if (paragraph?.type === 'paragraph') {
+    const linkChildren: Link[] = paragraph.children.filter(
+      (chart) => chart.type === 'link',
+    ) as Link[];
+    for (const linkChild of linkChildren) {
+      const link = linkChild?.url;
+      if (link) {
+        const label =
+          linkChild?.children[0]?.type === 'text'
+            ? linkChild.children[0].value
+            : undefined;
```

Comment:
no, you break KISS principle by this

Reply:
and its okey to have it like that

Reply:
What coderabbit proposed technically would be okay (and better) if our goal was to find the first child with "text" type. In this case it's assumed there will only be one root child because it's a link type.